src/tools/cephfs/Dumper.cc

```diff
@@ -108,13 +108,19 @@ int Dumper::dump(const char *dump_file)
     // include an informative header
     char buf[HEADER_LEN];
     memset(buf, 0, sizeof(buf));
-    snprintf(buf, HEADER_LEN, "Ceph mds%d journal dump\n start offset %llu (0x%llx)\n       length %llu (0x%llx)\n    write_pos %llu (0x%llx)\n    format %llu\n    trimmed_pos %llu (0x%llx)\n%c",
+    snprintf(buf, HEADER_LEN, "Ceph mds%d journal dump\n start offset %llu (0x%llx)\n\
+       length %llu (0x%llx)\n    write_pos %llu (0x%llx)\n    format %llu\n\
+    trimmed_pos %llu (0x%llx)\n    stripe_unit %lu (0x%lx)\n    stripe_count %lu (0x%lx)\n\
+    object_size %lu (0x%lx)\n    fsid %s\n%c",
```

Comment:
fsid left in from other PR. This causes a seg-fault:

````
(gdb) bt
#0  0x00007f4e6a3cb269 in raise (sig=sig@entry=11) at ../sysdeps/unix/sysv/linux/pt-raise.c:35
#1  0x000055b7f691d9b3 in reraise_fatal (signum=11) at /build/ceph-14.0.0-787-g8f48616/src/global/signal_handler.cc:80
#2  handle_fatal_signal (signum=11) at /build/ceph-14.0.0-787-g8f48616/src/global/signal_handler.cc:290
#3  <signal handler called>
#4  0x00007f4e69929cc0 in _IO_vfprintf_internal (s=s@entry=0x7fffd3b87d50, format=<optimized out>, 
    format@entry=0x55b7f6946008 "Ceph mds%d journal dump\n start offset %llu (0x%llx)\n       length %llu (0x%llx)\n    write_pos %llu (0x%llx)\n    format %llu\n    trimmed_pos %llu (0x%llx)\n    stripe_unit %lu (0x%lx)\n    stripe_count %"..., ap=ap@entry=0x7fffd3b87ec8) at vfprintf.c:1632
#5  0x00007f4e699f1896 in ___vsnprintf_chk (
    s=0x7fffd3b88770 "Ceph mds0 journal dump\n start offset 4194304 (0x400000)\n       length 31179 (0x79cb)\n    write_pos 4221504 (0x406a40)\n    format 1\n    trimmed_pos 4194304 (0x400000)\n    stripe_unit 4194304 (0x400000)"..., maxlen=<optimized out>, flags=1, slen=<optimized out>, 
    format=0x55b7f6946008 "Ceph mds%d journal dump\n start offset %llu (0x%llx)\n       length %llu (0x%llx)\n    write_pos %llu (0x%llx)\n    format %llu\n    trimmed_pos %llu (0x%llx)\n    stripe_unit %lu (0x%lx)\n    stripe_count %"..., args=args@entry=0x7fffd3b87ec8) at vsnprintf_chk.c:63
#6  0x00007f4e699f17f8 in ___snprintf_chk (
    s=s@entry=0x7fffd3b88770 "Ceph mds0 journal dump\n start offset 4194304 (0x400000)\n       length 31179 (0x79cb)\n    write_pos 4221504 (0x406a40)\n    format 1\n    trimmed_pos 4194304 (0x400000)\n    stripe_unit 4194304 (0x400000)"..., maxlen=maxlen@entry=4096, flags=flags@entry=1, slen=slen@entry=4096, 
    format=format@entry=0x55b7f6946008 "Ceph mds%d journal dump\n start offset %llu (0x%llx)\n       length %llu (0x%llx)\n    write_pos %llu (0x%llx)\n    format %llu\n    trimmed_pos %llu (0x%llx)\n    stripe_unit %lu (0x%lx)\n    stripe_count %"...) at snprintf_chk.c:34
#7  0x000055b7f65ddac0 in snprintf (
    __fmt=0x55b7f6946008 "Ceph mds%d journal dump\n start offset %llu (0x%llx)\n       length %llu (0x%llx)\n    write_pos %llu (0x%llx)\n    format %llu\n    trimmed_pos %llu (0x%llx)\n    stripe_unit %lu (0x%lx)\n    stripe_count %"..., __n=4096, 
    __s=0x7fffd3b88770 "Ceph mds0 journal dump\n start offset 4194304 (0x400000)\n       length 31179 (0x79cb)\n    write_pos 4221504 (0x406a40)\n    format 1\n    trimmed_pos 4194304 (0x400000)\n    stripe_unit 4194304 (0x400000)"...) at /usr/include/x86_64-linux-gnu/bits/stdio2.h:65
#8  Dumper::dump (this=this@entry=0x7fffd3b89980, dump_file=0x55b7f8934180 "/tmp/journal.bin") at /build/ceph-14.0.0-787-g8f48616/src/tools/cephfs/Dumper.cc:111
#9  0x000055b7f65c1c25 in JournalTool::journal_export (this=this@entry=0x7fffd3b89e40, path="/tmp/journal.bin", import=<optimized out>) at /build/ceph-14.0.0-787-g8f48616/src/tools/cephfs/JournalTool.cc:589
#10 0x000055b7f65c226b in JournalTool::main_journal (this=this@entry=0x7fffd3b89e40, argv=std::vector of length 2, capacity 3 = {...}) at /build/ceph-14.0.0-787-g8f48616/src/tools/cephfs/JournalTool.cc:207
#11 0x000055b7f65ba522 in JournalTool::main (this=0x7fffd3b89e40, argv=std::vector of length 2, capacity 3 = {...}) at /build/ceph-14.0.0-787-g8f48616/src/tools/cephfs/JournalTool.cc:168
#12 0x000055b7f65b97b6 in main (argc=<optimized out>, argv=<optimized out>) at /build/ceph-14.0.0-787-g8f48616/src/tools/cephfs/cephfs-journal-tool.cc:51
````
From: http://pulpito.ceph.com/pdonnell-2018-06-23_02:11:08-fs-wip-pdonnell-testing-20180622.235254-testing-basic-smithi/2692237/

Reply:
@batrick  so this means this pr needs rebased after that fsid-introduced pr is merged?

Reply:
will take a look.

Reply:
This can wait until the fsid PR goes in.